pkg/operator/cassandra/controller/service.go

```diff
@@ -0,0 +1,135 @@
+package controller
+
+import (
+	cassandrav1alpha1 "github.com/rook/rook/pkg/apis/cassandra.rook.io/v1alpha1"
+	"github.com/rook/rook/pkg/operator/cassandra/constants"
+	"github.com/rook/rook/pkg/operator/cassandra/controller/util"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/kubernetes/pkg/controller/endpoint"
+	"strings"
+)
+
+const (
+	unusedPort = 64551
+)
+
+// SyncClusterHeadlessService checks if a Headless Service exists
+// for the given Cluster, in order for the StatefulSets to utilize it.
+// If it doesn't exists, then create it.
+func (cc *ClusterController) SyncClusterHeadlessService(c *cassandrav1alpha1.Cluster) error {
+	clusterHeadlessService := &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            util.HeadlessServiceNameForCluster(c),
+			Namespace:       c.Namespace,
+			Labels:          util.ClusterLabels(c.Name),
+			OwnerReferences: []metav1.OwnerReference{util.NewControllerRef(c)},
+		},
+		Spec: corev1.ServiceSpec{
+			ClusterIP: corev1.ClusterIPNone,
+			Type:      corev1.ServiceTypeClusterIP,
+			Selector:  util.ClusterLabels(c.Name),
+			// Necessary to specify a Port to work correctly
+			// https://github.com/kubernetes/kubernetes/issues/32796
```

Comment:
this k8s issue is closed, was it not resolved upstream?

Reply:
It seems like it was, however it's not too clear in what version it was fixed. 
Since rook wants to maintain backwards compatibility to 1.8, I thought it would be prudent to leave it in just in case.